components/table/table-wrapper.js

```diff
@@ -94,26 +102,30 @@ export const tableStyles = css`
 	}
 
 	/* no-column-border */
+	d2l-table-wrapper[type="default"][no-column-border]:not([dir="rtl"]) .d2l-table > tbody > tr > *:not(.d2l-table-cell-last),
 	d2l-table-wrapper[type="default"]:not([dir="rtl"]) .d2l-table[no-column-border] > tbody > tr > *:not(.d2l-table-cell-last) {
 		border-right: none;
 	}
+	d2l-table-wrapper[type="default"][no-column-border][dir="rtl"] .d2l-table > tbody > tr > *:not(.d2l-table-cell-last),
 	d2l-table-wrapper[type="default"][dir="rtl"] .d2l-table[no-column-border] > tbody > tr > *:not(.d2l-table-cell-last) {
 		border-left: none;
 	}
 
 	/* sticky-headers */
 
 	/* all header cells */
-	d2l-table-wrapper[sticky-headers] .d2l-table > * > tr[header] > *,
-	d2l-table-wrapper[sticky-headers] .d2l-table > thead > tr > th {
+	d2l-table-wrapper[sticky-headers] .d2l-table > thead > tr > th,
+	d2l-table-wrapper[sticky-headers] .d2l-table > * > tr.d2l-table-header > *,
+	d2l-table-wrapper[sticky-headers] .d2l-table > * > tr[header] > * {
```

Comment:
Just flipped these around a bit so that it goes in order of goodness: `thead`, then the CSS class, then the legacy attribute.